base/pkg/entry.jl

```diff
@@ -197,7 +197,11 @@ function clone(url::AbstractString, pkg::AbstractString)
         end
     catch err
         isdir(pkg) && Base.rm(pkg, recursive=true)
-        rethrow(err)
+        if err.msg == "Unexpected HTTP status code: 404"
```

Comment:
will you get a 404 if you try installing a package with the correct url when your connection is disabled, or will the error be different?